sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -0,0 +1,615 @@
+    convertReadData<uint8_t>(
+        readPixel(reinterpret_cast<uint8_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unorm_int8, Color);
+    break;
+  case image_channel_type::unorm_int16:
+    convertReadData<uint16_t>(
+        readPixel(reinterpret_cast<uint16_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unorm_int16, Color);
+    break;
+  case image_channel_type::unorm_short_565:
+    convertReadData<ushort>(
+        readPixel(reinterpret_cast<ushort *>(Ptr), ImageChannelOrder),
+        image_channel_type::unorm_short_565, Color);
+    break;
+  case image_channel_type::unorm_short_555:
+    convertReadData<ushort>(
+        readPixel(reinterpret_cast<ushort *>(Ptr), ImageChannelOrder),
+        image_channel_type::unorm_short_555, Color);
+    break;
+  case image_channel_type::unorm_int_101010:
+    convertReadData<uint32_t>(
+        readPixel(reinterpret_cast<uint32_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unorm_int_101010, Color);
+    break;
+  case image_channel_type::signed_int8:
+    convertReadData<int8_t>(
+        readPixel(reinterpret_cast<int8_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::signed_int8, Color);
+    break;
+  case image_channel_type::signed_int16:
+    convertReadData<int16_t>(
+        readPixel(reinterpret_cast<int16_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::signed_int16, Color);
+    break;
+  case image_channel_type::signed_int32:
+    convertReadData<int32_t>(
+        readPixel(reinterpret_cast<int32_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::signed_int32, Color);
+    break;
+  case image_channel_type::unsigned_int8:
+    convertReadData<uint8_t>(
+        readPixel(reinterpret_cast<uint8_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unsigned_int8, Color);
+    break;
+  case image_channel_type::unsigned_int16:
+    convertReadData<uint16_t>(
+        readPixel(reinterpret_cast<uint16_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unsigned_int16, Color);
+    break;
+  case image_channel_type::unsigned_int32:
+    convertReadData<uint32_t>(
+        readPixel(reinterpret_cast<uint32_t *>(Ptr), ImageChannelOrder),
+        image_channel_type::unsigned_int32, Color);
+    break;
+  case image_channel_type::fp16:
+    convertReadData<cl_half>(
+        readPixel(reinterpret_cast<cl_half *>(Ptr), ImageChannelOrder),
+        image_channel_type::fp16, Color);
+    break;
+  case image_channel_type::fp32:
+    convertReadData<float>(
+        readPixel(reinterpret_cast<float *>(Ptr), ImageChannelOrder),
+        image_channel_type::fp32, Color);
+    break;
+  }
+
+  return Color;
+}
+
+// imageReadSamplerHostImpl method is called by the read API in image accessors
+// for host device.
+// Algorithm used:
+// Step 1.
+//   get u,v,w coordinates for processing and getting pixel Coordinates.
+//   u,v,w are normalized for AddrMode:mirror_repeat and repeat.
```

Comment:
It is because the spec methodolgy is such. 
It uses normalised coordinates for repeat/mirror_repeat and unnormalised coordinates for clamp/clamp_to_edge/none Addr Mode, to compute the coordinates to load data from. 
It is given in spec 8.3